vscode/src/lsp/nbLanguageClient.ts

```diff
@@ -0,0 +1,103 @@
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { LanguageClient, LanguageClientOptions, ServerOptions } from 'vscode-languageclient/node';
+import { CloseAction, CloseHandlerResult, DocumentSelector, ErrorAction, ErrorHandlerResult, Message, RevealOutputChannelOn } from "vscode-languageclient";
+import { createTreeViewService, TreeViewService } from "../views/projects";
+import { OutputChannel, workspace } from "vscode";
+import { extConstants } from "../constants";
+import { userConfigsListenedByServer } from '../configurations/configuration';
+import { restartWithJDKLater } from './utils';
+import { ExtensionLogger } from '../logger';
+import { globalVars } from '../extension';
+
+
+export class NbLanguageClient extends LanguageClient {
+    private _treeViewService: TreeViewService;
+    constructor(id: string, name: string, s: ServerOptions, log: OutputChannel, c: LanguageClientOptions) {
+        super(id, name, s, c);
+        this._treeViewService = createTreeViewService(log, this);
+    }
+
+    static build = (serverOptions: ServerOptions, logger: ExtensionLogger): NbLanguageClient => {
+        let documentSelectors: DocumentSelector = [
+            { language: extConstants.LANGUAGE_ID },
+            { language: 'properties', pattern: '**/*.properties' },
+            { language: 'jackpot-hint' },
+            { language: 'xml', pattern: '**/pom.xml' },
+            { pattern: '*.gradle' },
+            { pattern: '*.gradle.kts' }
+        ];
+
+        // Options to control the language client
+        let clientOptions: LanguageClientOptions = {
+            // Register the server for java documents
+            documentSelector: documentSelectors,
+            synchronize: {
+                configurationSection: userConfigsListenedByServer,
+                fileEvents: [
+                    workspace.createFileSystemWatcher('**/*.java')
+                ]
+            },
+            outputChannel: logger.getOutputChannel(),
+            revealOutputChannelOn: RevealOutputChannelOn.Never,
+            progressOnInitialization: true,
+            initializationOptions: {
+                'nbcodeCapabilities': {
+                    'statusBarMessageSupport': true,
+                    'testResultsSupport': true,
+                    'showHtmlPageSupport': true,
+                    'wantsJavaSupport': true,
+                    'wantsGroovySupport': false,
+                    'commandPrefix': extConstants.COMMAND_PREFIX,
+                    'configurationPrefix': 'jdk.',
+                    'altConfigurationPrefix': 'jdk.'
```

Comment:
would this also be `extConstants.COMMAND_PREFIX`?

Reply:
Yes, updated